sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/models/SqlQuerySpecWithEncryption.java

```diff
@@ -98,6 +119,13 @@ Mono<Void> addEncryptionParameterAsync(String path, SqlParameter sqlParameter,
         return Mono.empty();
     }
 
+    private String convertToBase64UriSafeString(byte[] bytesToProcess) {
+        // Base 64 Encoding with URL and Filename Safe Alphabet  https://datatracker.ietf.org/doc/html/rfc4648#section-5
+        // https://docs.microsoft.com/en-us/azure/cosmos-db/concepts-limits#per-item-limits, due to base64 conversion and encryption
+        // the permissible size of the property will further reduce.
+        return Base64.getUrlEncoder().encodeToString(bytesToProcess);
```

Comment:
This doesn't seem to be the same encoding .Net is using? Please make sure and test that .Net is using compatible encoding - it must be possible to write encrypted doc with .Net and read via java and vice versa.

Reply:
In java, we directly have a URL encoder that does the same encoding that .Net is doing. 

In .Net :
- Convert to base 64 string
- Replace + to -
- Replace / to _

The Jave URL encoder directly does that encoding which is URL safe, as mentioned in this link.
 https://datatracker.ietf.org/doc/html/rfc4648#section-5

Reply:
This looks good to me, thanks for the explanation @aayush3011